File: src/main/java/com/todolist/task/service/application/mapper/TaskDTOMapper.java
package com.todolist.task.service.application.mapper;

import com.todolist.task.service.application.models.TaskDTO;
import com.todolist.task.service.domain.models.Task;
import java.util.List;
import org.mapstruct.Mapper;
import org.mapstruct.ReportingPolicy;
import org.mapstruct.factory.Mappers;

@Mapper(unmappedTargetPolicy = ReportingPolicy.IGNORE)
public abstract class TaskDTOMapper {

  public static final TaskDTOMapper MAPPER = Mappers.getMapper(
      TaskDTOMapper.class);

  public abstract List<TaskDTO> toDTO(List<Task> tasks);

}

File: src/main/kotlin/com/todolist/task/service/application/mapper/TaskDTOMapper.kt
package com.todolist.task.service.application.mapper

import com.todolist.task.service.application.models.TaskDTO
import com.todolist.task.service.domain.models.Task

abstract class TaskDTOMapper {
    companion object {
        fun toDTO(task: Task): TaskDTO {
            return TaskDTO(
                id = task.id,
                title = task.title,
                description = task.description,
                priority = task.priority,
                status = task.status,
                dueDate = task.dueDate
            )
        }
    }
}
